index.js

```diff
@@ -17,6 +17,13 @@ var gs = {
     // remove path relativity to make globs make sense
     ourGlob = unrelative(opt.cwd, ourGlob);
 
+    if (negatives.length) {
+      // Do not mutate the options object which may be used
+      // in multiple `gs.createStream()` calls.
+      opt = assign({}, opt);
```

Comment:
I tend to use `object-assign` as its logic is rather simple and well-spec'd (follows ECMAScript 2015 spec.). But yes, I guess a clone or cloneDeep would suffice as well.